### PR TITLE
Refactor code base

### PR DESCRIPTION
**Pull request by kylebonnici**

### CODE REVIEW

**Comment by datenreisender** [comment]
What is the goal of this refactoring?

**Comment by kylebonnici** [comment]
@datenreisender, goal was to pass time as I was not able to sleep xD so I decided to update this code to make it more readable and more easy to follow.

**Comment by kylebonnici** [comment]
Also this uses the latest device setup from shared

File: src/actions/deviceActions.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            {
+                key: 'nrf52_family',
+                fw: getAppFile('fw/rssi-10040.hex'),
+                fwVersion: 'rssi-fw-1.0.0',
+                fwIdAddress: 0x2000,
+            },
+        ]),
+    ],
+    needSerialport: true,
+};
```
Comment (PascalPickel) [inline src/actions/deviceActions.ts:46]: If this is here only because of the recoverHex, I'd prefer to have it in a separate file.

**Pull request merged**